pkg/bootstrap/config/types.go

```diff
@@ -124,3 +131,56 @@ type SecretsSetupInfo struct {
 	// CertOutputDir indicates the folder for auto-generated TLS certificates
 	CertOutputDir string
 }
+
+// ClientsCollection is a collection of Client information for communicating to dependent clients.
+type ClientsCollection map[string]*ClientInfo
+
+// ClientInfo provides the host and port of another service in the eco-system.
+type ClientInfo struct {
+	// Host is the hostname or IP address of a service.
+	Host string
+	// Port defines the port on which to access a given service
+	Port int
+	// Protocol indicates the protocol to use when accessing a given service
+	Protocol string
+	// UseMessageBus indicates weather to use Messaging version of client
+	UseMessageBus bool
+	// SecurityOptions is a key/value map, used for configuring clients. Currently used for zero trust but
+	// could be for other options additional security related configuration
+	SecurityOptions map[string]string
+}
+
+func (c ClientInfo) Url() string {
+	url := fmt.Sprintf("%s://%s:%v", c.Protocol, c.Host, c.Port)
+	return url
+}
+
+func NewSecretStoreSetupClientInfo() *ClientsCollection {
+	secretStoreStepClient := ClientsCollection{
+		SecuritySecretStoreSetupServiceKey: &ClientInfo{
+			Host:     "localhost",
+			Port:     59843,
+			Protocol: "http",
+		}}
+	return &secretStoreStepClient
+}
+
+func NewSecretStoreInfo(serviceKey string) SecretStoreInfo {
+	return SecretStoreInfo{
+		Type:                    client.DefaultSecretStore,
+		Protocol:                "http",
+		Host:                    "localhost",
+		Port:                    8200,
+		StoreName:               serviceKey,
+		TokenFile:               fmt.Sprintf("/tmp/edgex/secrets/%s/secrets-token.json", serviceKey),
```

Comment:
pleasse search and examine all the edgex related part
the token filepath should not be hardcoded

Reply:
It can be overwritten by the environment variable `SECRETSTORE_TOKENFILE`
https://github.com/IOTechSystems/go-mod-edge-utils/pull/76/files#diff-e57fdc1d97a5595e77bbfee7d741ecff33822b681fc38dd687f311660e3d1cc0R175-R191